appinfo/app.php

```diff
@@ -4,3 +4,18 @@
 
 $app = new \OCA\SocialLogin\AppInfo\Application();
 $app->register();
+
+$userSession = \OC::$server->getUserSession();
+$config = \OC::$server->getConfig();
+
+if(!$userSession->isLoggedIn() &&
+	\OC::$server->getRequest()->getPathInfo() === '/login') {
+		$autoRedirect = $config->getAppValue('sociallogin', 'auto_redirect');
+		$alternativeLogins = \OC_App::getAlternativeLogins();
+
+		if($autoRedirect && count($alternativeLogins)==1){
```

Comment:
Alternative logins can be not only from social login app. https://apps.nextcloud.com/apps/registration use it too for example.

Reply:
Indeed. I can move the checkbox somewhere else, under "Additional settings" for instance.

Reply:
No. I think it should be related to this app. Auto redirect to register page is nonsence. 
It may be if register app is installed and checkbox is checked without configured providers.
Maybe I merge it as is and modify by my visions...
 
And I don't like using of `\OC` - it is "private" and can be deprecated or removed in future I think. 
`\OC_App::*AlternativeLogins` is exception coz developers just forgot about it and don't implement such methods in `\OCP\App`

Reply:
And checking for '/login', `header` function and `exit` looks like a hack. I like idea, but don't like implementation.

Reply:
But it simple 😏 And coz of it should contains less errors )

Reply:
I took the idea from here: 
https://github.com/nextcloud/user_saml/blob/master/appinfo/app.php#L96

It looks like a hack indeed but if the official team does it...

Reply:
That code from "some" app. It is not normal, but... Maybe no other ways.